doc/concepts/morphology.rst

```diff
@@ -482,6 +482,96 @@ because it has two children: the dendrites attached to its distal end.
+`SWC <http://www.neuronland.org/NLMorphologyConverter/MorphologyFormats/SWC/Spec.html>`_ file format.
+SWC files describe the morphology as a list of samples with an id, an `x,y,z` location is space, a radius, a tag
+and a parent id. Arbor parses these samples, performs some checks, then generates a
+:ref:`segment tree <morph-segment_tree>` describing the morphology according to one of three possible
+interpretations.
+
+The SWC file format specifications are not very detailed, which has lead different simulators to interpret
+SWC files in different ways, especially when it comes to the soma. Arbor has its own an interpretation that
+is powerful, and simple to understand at the same time. However, we have also developed functions that will
+interpret SWC files similarly to how the NEURON simulator would, and how the Allen Institute would.
+
+Despite the differences between the interpretations, there is a common set of checks that are always performed
+to check the validity of SWC files:
+   * Check that there are no duplicate ids.
+   * Check that the parent id of a sample is less than the id of the sample.
+   * Check that the parent id of a sample refers to an existing sample.
+
+In addition, all interpretations agree that a *segment* is (in the common case) constructed between a sample and
+its parent sample; and if more than 1 sample have the same parent sample, the parent sample is interpreted as a fork
+point in the morphology, and acts as the proximal point to a new branch for each of its "child" samples.
+There a couple of exceptions to these rules which are listed below.
+
+Arbor interpretation:
+"""""""""""""""""""""
+In addition to the previously listed checks, the arbor interpretation explicitly disallows SWC files where the soma is
+described by a single sample. It constructs the soma from 2 or more samples that form 1 or more segments in the segment
+tree. A *segment* is constructed between a sample and its parent, and inherits the sample's tag, regardless of
+whether or not its tag matches the parent's tag. This means that there are no gaps in the resulting segment tree.
+
+Arbor has no magic rules or transformations for the soma. It can be a single branch or multiple branches; segments
+of a different tag can connect to its distal end, proximal end or anywhere in the middle. For example, to create a
+morphology with a single segment soma; a single segment axon connected to one end of the soma; and a single segment
+dendrite connected to the other end of the soma, the following swc file can be used:
+
+.. code:: Python
+
+   # id, tag, x, y,   z, r, parent
+      1,   1, 0, 0,   0, 5, -1
+      2,   1, 0, 0,  10, 5,  1
+      3,   2, 0, 0, -10, 2,  1
+      4,   3, 0, 0, 100, 1,  2
+
+Samples 1 and 2 will form the soma; samples 1 and 3 will form the axon, connected to the soma at the proximal end;
+samples 2 and 4 will form the dendrite, connected to the soma at the distal end.
```

Comment:
I would love to see Ben's sketches here.

Reply:
I considered that but decided it wasn't really necessary. But I can add one if you like.

Reply:
I think it helps, but maybe just refer/link to the single cell docs?